pymc3/distributions/continuous.py

```diff
@@ -924,6 +925,73 @@ def logcdf(self, value):
         )
 
 
+class ZeroSumNormal(Continuous):
+    def __new__(cls, name, *args, **kwargs):
+        zerosum_axes = kwargs.get("zerosum_axes", None)
+        zerosum_dims = kwargs.get("zerosum_dims", None)
+        dims = kwargs.get("dims", None)
+
+        if isinstance(zerosum_dims, str):
+            zerosum_dims = (zerosum_dims,)
+        if isinstance(dims, str):
+            dims = (dims,)
+
+        if zerosum_dims is not None:
+            if dims is None:
+                raise ValueError("zerosum_dims can only be used with the dims kwargs.")
+            if zerosum_axes is not None:
+                raise ValueError("Only one of zerosum_axes and zerosum_dims can be specified.")
+            zerosum_axes = []
+            for dim in zerosum_dims:
+                zerosum_axes.append(dims.index(dim))
+            kwargs["zerosum_axes"] = zerosum_axes
+
+        return super().__new__(cls, name, *args, **kwargs)
+
+    def __init__(self, sigma=1, zerosum_axes=None, zerosum_dims=None, **kwargs):
+        shape = kwargs.get("shape", ())
+        if isinstance(shape, int):
+            shape = (shape,)
+
+        self.mu = self.median = self.mode = tt.zeros(shape)
+        self.sigma = tt.as_tensor_variable(sigma)
+
+        if zerosum_axes is None:
+            if shape:
+                zerosum_axes = (-1,)
+            else:
+                zerosum_axes = ()
+
+        if isinstance(zerosum_axes, int):
+            zerosum_axes = (zerosum_axes,)
+
+        self.zerosum_axes = [a if a >= 0 else len(shape) + a for a in zerosum_axes]
```

Comment:
Enforcing positive axis here leads to problems when you draw samples from the prior predictive. It's better to replace this line with this
```suggestion
        self.zerosum_axes = [a if a < 0 else a - len(shape) for a in zerosum_axes]
```